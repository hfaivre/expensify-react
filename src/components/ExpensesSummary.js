import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
import "numeral/locales/fr";


numeral.locale('fr');

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
	const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
	const formattedExpensesTotal = numeral(expensesTotal/100).format('$0,0.00');
	return (
		<div>
			<h1>Viewing {expenseCount} {expenseWord} totalling  {formattedExpensesTotal}</h1>
		</div>
	)
};
	



const mapStateToProps = (state) => {
  return {
    expenseCount: selectExpenses(state.expenses, state.filters).length,
    expensesTotal : expensesTotal(selectExpenses(state.expenses, state.filters))
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
