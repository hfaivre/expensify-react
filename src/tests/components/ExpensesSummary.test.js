import React from 'react';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';
import {ExpensesSummary} from '../../components/ExpensesSummary';


test('Should render Expenses Summary with expense count of 1',()=>{
	const wrapper = shallow( <ExpensesSummary expenseCount = {1} expensesTotal = {109500}/>);
	expect(wrapper).toMatchSnapshot();
});

test('Should render Expenses Summary with expense count of 1++',()=>{
	const wrapper = shallow( <ExpensesSummary expenseCount = {3} expensesTotal = {114195}/>);
	expect(wrapper).toMatchSnapshot();
});