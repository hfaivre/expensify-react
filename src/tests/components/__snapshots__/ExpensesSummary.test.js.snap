// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expenses Summary with expense count of 1 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling  
    €1 095,00
  </h1>
</div>
`;

exports[`Should render Expenses Summary with expense count of 1++ 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling  
    €1 141,95
  </h1>
</div>
`;
